qiskit/primitives/base_result.py

```diff
@@ -0,0 +1,62 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Primitive result abstract base class
+"""
+
+from abc import ABC
+from dataclasses import fields
+from typing import Any, Sized, Tuple
+
+
+class BaseResult(ABC):
```

Comment:
`BasePrimitiveResult` might be more specific as a base result class for primitives though it's not a strong opinion.